### PR TITLE
Molecular Weight Sampler

### PR DESCRIPTION
**Pull request by kspieks**
This PR introduces support for an extrapolative sampler that splits molecules by molecular weight. One option to create more challenging splits is to train ML models on smaller molecules and then test performance on larger molecules to determine whether the model has actually learned certain chemical intuition about mapping substructures to properties rather than just memorizing the training set.

In accordance with this approach, the new `MolecularWeight` sampler first sorts the molecules by molecular weight and then places the smallest molecules in the training set, the next smallest in the validation set if applicable, and finally the largest molecules in the testing set. 

To document a few general notes,

• This sampler sorts the indices by molecular weight. Since no clusters are created, the `MolecularWeight` class actually behaves as an interpolation sampler behind the scenes when it's called in `main.py`. 
• The other consequence of sorting is that the results are always deterministic. This sampler does not use the `random_state` argument. If users want to obtain multiple splits, they can vary the amount of data that is assigned to the training vs. testing set

Let me know if I missed anything. Looking forward to your feedback!

### CODE REVIEW

**Comment by kspieks** [comment]
No worries at all! Hope your work is going well. Yes please feel encouraged
to push commits. Sorry I wasn’t able to finalize the formatting last
weekend. It’s been a busy week for me as well.

**Comment by JacksonBurns** [comment]
Hi @kspieks thanks for your patience on this while I dug myself out of a mountain of back work, as per usual.

I have been thinking about this PR and #170 and realized this sampler would actually be a subclass of that one. I am going to pull your commits from these PRs into a unified branch and we can get this merged!

**Comment by kspieks** [comment]
That's a great suggestion. Thanks for refactoring the code!
